src/Microsoft.OData.Client/Serialization/HttpClientRequestMessage.cs

```diff
@@ -39,18 +40,22 @@ public class HttpClientRequestMessage : DataServiceClientRequestMessage, ISendin
             "Expires",
             "Last-Modified",
         }, StringComparer.OrdinalIgnoreCase);
-
-        /// <summary>Request Url.</summary>
-        private readonly Uri _requestUrl;
+        /// <summary>
+        /// HttpClient to use when the caller does not provide one.
+        /// </summary>
+        private static readonly HttpClient _defaultClient = new HttpClient();

```

Comment:
```suggestion

        /// <summary>
        /// HttpClient to use when the caller does not provide one.
        /// </summary>
        private static readonly HttpClient _defaultClient = new HttpClient();
```

Reply:
Shouldn't the factory be used to create the client here?

Reply:
Or why are we creating a HttpClient beforehand instead of doing it when factory is not provided? Why can't OData CLient use the IHttpClientFactory to create httpclients instead of newing them directly?

Reply:
This is the "default" client that will be used just in case the user does not provide an `IHttpClientFactory`. If they provide one, we don't use this. If they don't provide one, then we don't have an `IHttpClientFactory` to use, and so we default to this one.

Note that we're not creating the `HttpClient` "beforehand". This is a static field initialization, the runtime will only create the client if the `_defaultClient` field is accessed. This means that if the application always provides an `IHttpCientFactory` then the default `HttpClient` will never be allocated.
 
See how it's used here: https://github.com/OData/odata.net/pull/2883/files#diff-7ffa90011377965c2da0275b44b7914818cf3aacaaaabdd766231a449f3ad656R88